File: TombolaJava/src/tombola/Tombola.java
package tombola;

import java.util.Random;

import it.unical.mat.embasp.base.Handler;
import it.unical.mat.embasp.base.InputProgram;
import it.unical.mat.embasp.base.OptionDescriptor;
import it.unical.mat.embasp.languages.asp.ASPInputProgram;
import it.unical.mat.embasp.languages.asp.ASPMapper;
import it.unical.mat.embasp.languages.asp.AnswerSet;
import it.unical.mat.embasp.languages.asp.AnswerSets;
import it.unical.mat.embasp.platforms.desktop.DesktopHandler;
import it.unical.mat.embasp.specializations.dlv2.desktop.DLV2DesktopService;

public class Tombola {

	public static void main(String[] args) {

		try {
			Handler handler = new DesktopHandler(new DLV2DesktopService("lib/dlv2"));

			ASPMapper.getInstance().registerClass(Casella.class);

			InputProgram input = new ASPInputProgram();
			input.addFilesPath("src/tombola/codice");

			handler.addProgram(input);
			handler.addOption(new OptionDescriptor("-n 1000"));

			AnswerSets answer = (AnswerSets) handler.startSync();
			
			int count = 0;
			for(AnswerSet a : answer.getAnswersets()) {
				count++;
			}
			
			int random = new Random().nextInt(count);
			
			Cartella cartella = new Cartella();
			
			count = -1;
			for(AnswerSet a : answer.getAnswersets()){
				count++;
				if(count == random) {
					for(Object obj : a.getAtoms()) {
						if(obj instanceof Casella) {
							Casella c = (Casella) obj;
							cartella.set(c.getRiga(), c.getColonna(), c.getValore());
						}
					}					
				}
			}

			cartella.stampa();
			
		} catch (Exception e) {
			e.printStackTrace();
		} 
	}

}

File: TombolaPython/src/tombola.py
from languages.predicate import Predicate
from languages.asp.asp_mapper import ASPMapper
from platforms.desktop.desktop_handler import DesktopHandler
from specializations.dlv2.desktop.dlv2_desktop_service import DLV2DesktopService
from languages.asp.asp_input_program import ASPInputProgram

class Cartella:
    def __init__(self):
        self.caselle = ["  "] * 3
        for i in range(3):
            self.caselle[i] = ["  "] * 9
        
    def set(self, riga, colonna, valore):
        self.caselle[(int(riga))-1][(int(colonna))-1] = valore
        
    def stampa(self):
        for riga in range(3):
            for colonna in range(9):
                print(self.caselle[riga][colonna], end=' | ')
            print("")
            
class Casella(Predicate):
    predicate_name = "casella"
    
    def __init__(self, riga=None, colonna=None, valore=None):
        Predicate.__init__(self, [("riga"),("colonna"),("valore")])
        self.riga = riga
        self.colonna = colonna
        self.valore = valore
        
    def get_riga(self):
        return self.riga
    
    def get_colonna(self):
        return self.colonna
    
    def get_valore(self):
        return self.valore
    
    def set_riga(self, riga):
        self.riga = riga
        
    def set_colonna(self, colonna):
        self.colonna = colonna
        
    def set_valore(self, valore):
        self.valore = valore
                

ASPMapper.get_instance().register_class(Casella)

inputProgram = ASPInputProgram()
inputProgram.add_files_path("./codice")

handler = DesktopHandler(DLV2DesktopService("../lib/dlv2"))
handler.add_program(inputProgram)

out = handler.start_sync()

cartella = Cartella()

for ans in out.get_answer_sets():
    for obj in ans.get_atoms():
        if isinstance(obj, Casella):
            cartella.set(obj.get_riga(), obj.get_colonna(), obj.get_valore())
        
cartella.stampa()




